README.md

```diff
@@ -86,6 +87,14 @@ Learning about the Substra platform:
 - [Cross-validation](./examples/cross_val/README.md)
 - [Compute plan](./examples/compute_plan/README.md)
 
+## Compatibility table
+
+These sets of versions have been tested for compatilibility: all the tests from [`substra-tests`](https://github.com/SubstraFoundation/substra-tests/) pass.
+
+| Major release?  | substra  | substra-chaincode  | substra-backend  | substra-tests  |
```

Comment:
I would remove the "major release" column as Samuel said and add the substra-frontend and hlf-k8s to the table.

Reply:
Done.

substra-frontend is a bit trickier, because it doesn't have end to end tests. As a result, one can easily assume that the front end "works" after loading a few pages, when actually some other page is broken because of a recent backend/chaincode change. Adding a "not fully tested" substra-frontend version to the compatibility table would defeat the purpose of the compatibility table.

To mitigate this, I added a (*) linking to (yet to be written) frontend testing instructions, and instructing to leave the column blank unless the frontend has been fully tested.

Happy to have your thoughts!

Reply:
Looks good to me.

As far as I understand, we will add a new line to this table each time we release one of our "services". Is it correct? 

So for each new release, we should at least ensure the frontend is not broken.

Reply:
> As far as I understand, we will add a new line to this table each time we release one of our "services". Is it correct?

I'm not sure which guidelines or rules we should have for adding a new row. Major releases seem like a minimum. We could do for minor releases too, or even commits without releases if we haven't released anything in a while :man_shrugging: I'm not sure. As long as we test thoroughly before adding a row, we could add rows pretty often!

> So for each new release, we should at least ensure the frontend is not broken.

Agree! However, in the absence of instructions, I don't think we currently have a way to do that reliably (except by people who are very familiar with the frontend and its testing @jmorel :eyes:)

Reply:
> As a result, one can easily assume that the front end "works" after loading a few pages, when actually some other page is broken because of a recent backend/chaincode change

I guess you don't test every function call or even all the API endpoints neither so that's ok for me to include the frontend even if we are not 100% sure: it will, for sure, help create a QA checklist on this repo too.

Reply:
EDIT: if you provide a test protocol for the frontend you should provide one for the other components (even if the protocol ends up being "launched this command and it returned 0") so everyone knows what to expect.

Reply:
Good points!

Reply:
https://github.com/SubstraFoundation/substra/pull/95/commits/a75c6dc561f502577a71081167bf492fee1d7c23 Does that look better?